axlearn/common/checkpointer_test.py

```diff
@@ -559,17 +559,27 @@ def test_read_state_spec(self):
 
 
 class TensorStoreStateStorageTest(test_utils.TestCase):
-    @parameterized.parameters(jnp.float32, jnp.bfloat16, jnp.int32, jnp.int16)
-    def test_save_and_restore_from_dir(self, restore_floats_as: jnp.dtype):
+    @parameterized.parameters(
+        [jnp.float32, None],
+        [jnp.bfloat16, None],
+        [jnp.int32, None],
+        [jnp.int16, None],
+        [jnp.float32, jnp.bfloat16],
+        [jnp.float32, jnp.float16],
+    )
+    def test_save_and_restore_from_dir(self, restore_floats_as: jnp.dtype, cast_dtype: jnp.dtype):
```

Comment:
Comment on the difference between `restore_floats_as` and `cast_dtype`? Shouldn't they be the same?